src/uu/chmod/src/chmod.rs

```diff
@@ -18,115 +19,173 @@ use uucore::fs::display_permissions_unix;
+            // however .multiple(true) cannot be used here because FILE already needs that.
+            // Only one positional argument with .multiple(true) set is allowed per command
+        )
+        .arg(
+            Arg::with_name(options::FILE)
+                .required_unless(options::MODE)
+                .multiple(true),
+        )
+        .get_matches_from(args);
+
+    let changes = matches.is_present(options::CHANGES);
+    let quiet = matches.is_present(options::QUIET);
+    let verbose = matches.is_present(options::VERBOSE);
+    let preserve_root = matches.is_present(options::PRESERVE_ROOT);
+    let recursive = matches.is_present(options::RECURSIVE);
+    let fmode =
+        matches
+            .value_of(options::REFERENCE)
             .and_then(|ref fref| match fs::metadata(fref) {
                 Ok(meta) => Some(meta.mode()),
                 Err(err) => crash!(1, "cannot stat attributes of '{}': {}", fref, err),
             });
-        let cmode = if fmode.is_none() {
-            // If there was a negative option, now it's a good time to
-            // use it.
-            if negative_option.is_some() {
-                negative_option
-            } else {
-                Some(matches.free.remove(0))
-            }
-        } else {
-            None
-        };
-        let chmoder = Chmoder {
-            changes,
-            quiet,
-            verbose,
-            preserve_root,
-            recursive,
-            fmode,
-            cmode,
-        };
-        match chmoder.chmod(matches.free) {
-            Ok(()) => {}
-            Err(e) => return e,
-        }
+    let modes = matches.value_of(options::MODE).unwrap(); // should always be Some because required
+    let mut cmode = if mode_had_minus_prefix {
+        // clap parsing is finished, now put prefix back
+        Some(format!("-{}", modes))
+    } else {
+        Some(modes.to_string())
+    };
+    let mut files: Vec<String> = matches
+        .values_of(options::FILE)
+        .map(|v| v.map(ToString::to_string).collect())
+        .unwrap_or_default();
+    if fmode.is_some() {
+        // "--reference" and MODE are mutually exclusive
+        // if "--reference" was used MODE needs to be interpreted as another FILE
+        // it wasn't possible to implement this behavior directly with clap
+        files.push(cmode.unwrap());
+        cmode = None;
+    }
+
+    let chmoder = Chmoder {
+        changes,
+        quiet,
+        verbose,
+        preserve_root,
+        recursive,
+        fmode,
+        cmode,
+    };
+    match chmoder.chmod(files) {
+        Ok(()) => {}
+        Err(e) => return e,
     }
 
     0
 }
 
-fn sanitize_input(args: &mut Vec<String>) -> Option<String> {
+// Iterate 'args' and delete the first occurrence
+// of a prefix '-' if it's associated with MODE
+// e.g. "chmod -v -xw -R FILE" -> "chmod -v xw -R FILE"
+fn strip_minus_from_mode(args: &mut Vec<String>) -> bool {
```

Comment:
this function would need some unit test at the function level

Reply:
done